tests/e2e/vsan_max_tkg_wldi.go

```diff
@@ -0,0 +1,315 @@
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, item := range eventList.Items {
+			framework.Logf("%q", item.Message)
+		}
+	})
+
+	/*
+		Testcase-5 & 6
+		vSAN Max with Fault Domains with HCI mounted datastore - Block & file Volume
+
+		Steps:
+		1. Deploy statefulsets with 3 replica on namespace-4 in the TKG cluster
+		1.1 Use vsan-shared policy with WFFC volume binding mode storageclass 1
+		1.2 Access Mode as ReadWriteOnce and ReadWriteMany
+		2. Bring down one of the host from vsphere cluster in zone-2
+		3. Bring down one of the host from vSAN max fault domain-1
+		4. Bring up host from zone-2 and host from fault domain-1
+		5. Bring down both hosts from vsphere cluster in zone-3
+		6. Bring down both hosts from vSAN max fault domain-2
+		7. Bring up all hosts from zone-3 and all hosts from fault domain-2
+	*/
+
+	ginkgo.It("vSAN Max with Fault Domains with HCI mounted datastore", ginkgo.Label(p0, wldi, vc90u1), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// get the vsan-max FD name
+		vsanMaxFdName := GetAndExpectStringEnvVar(vsanMaxFaultDomainName)
+
+		// statefulset replica count
+		replicas = 3
+
+		// Get the fault domain and host map
+		fdMap := createFaultDomainMap(ctx, &e2eVSphere)
+
+		// Flag to check the status of hosts
+		isHostDown := false
+		isFdHostDown := false
+
+		ginkgo.By("Read shared storage policy tagged to wcp namespace")
+		storageclass, err := client.StorageV1().StorageClasses().Get(ctx, sharedStoragePolicyNameWffc, metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+
+		ginkgo.By("Creating statefulset with ReadWriteOnce")
+		statefulsetRwo := createCustomisedStatefulSets(ctx, client, namespace, true, replicas, true, allowedTopologies,
+			true, true, "", "", storageclass, storageclass.Name)
+		defer func() {
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
+		}()
+
+		ginkgo.By("Verify svc pv affinity, pvc annotation and pod node affinity")
+		err = verifyPvcAnnotationPvAffinityPodAnnotationInSvc(ctx, client, statefulsetRwo, nil, nil, namespace,
+			allowedTopologies)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating statefulset with ReadWriteMany")
+		statefulsetRwm := createCustomisedStatefulSets(ctx, client, namespace, true, replicas, true, allowedTopologies,
+			true, true, "", v1.ReadWriteMany, storageclass, storageclass.Name)
+		defer func() {
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
+		}()
+
+		ginkgo.By("Verify svc pv affinity, pvc annotation and pod node affinity")
+		err = verifyPvcAnnotationPvAffinityPodAnnotationInSvc(ctx, client, statefulsetRwm, nil, nil, namespace,
+			allowedTopologies)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Get all hosts of zone2
```

Comment:
Can you write a util and put all these steps 181 till 193 in some util, we can then call a util, even i need the same logic.. pass all parameters and fetch all the values